solc/CommandLineInterface.cpp

```diff
@@ -807,10 +807,15 @@ Allowed options)",
 
 bool CommandLineInterface::processInput()
 {
-	ReadCallback::Callback fileReader = [this](string const& _path)
+	ReadCallback::Callback fileReader = [this](string const& _kind, string const& _path)
 	{
 		try
 		{
+			if (_kind != ReadCallback::kindString(ReadCallback::Kind::ReadFile))
```

Comment:
What case can this happen? Why not return an error message and not an exception?

Reply:
Only internally in case someone coded it wrong. Same idea as `solAssert`